### PR TITLE
Migrate to static interop.

### PR DESCRIPTION
**Pull request by joshualitt**
This CL migrates pkg/test to static interop.

### CODE REVIEW

File: pkgs/test/test/runner/browser/code_server.dart [hunk 1]
```diff
@@ -35,31 +35,6 @@ class CodeServer {
 
   CodeServer._(this._handler, this.url);
 
-  /// Sets up a handler for the root of the server, "/", that serves a basic
-  /// HTML page with a script tag that will run [dart].
-  void handleDart(String dart) {
```
Comment (jakemac53) [inline pkgs/test/test/runner/browser/code_server.dart:40]: Was this just unused? Curious how you were able to determine that?

File: pkgs/test/lib/src/runner/browser/static/host.dart.js [hunk 2]
```diff
@@ -6190,171 +6123,329 @@
       _.path = t4;
       _._query = t5;
       _._fragment = t6;
-      _.___Uri_queryParameters = _.___Uri_hashCode = _.___Uri_pathSegments = _.___Uri__text = $;
+      _.___Uri_queryParameters_FI = _.___Uri_hashCode_FI = _.___Uri_pathSegments_FI = _.___Uri__text_FI = $;
     },
-    window() {
-      return window;
+    HtmlElement: function HtmlElement() {
```
[the remark this answers is not in the document]
Comment (joshualitt) [reply pkgs/test/lib/src/runner/browser/static/host.dart.js:6128]: I meant to add a comment. I actually talked about this with Siggi last night when I noticed the size increase. I guess Dart2js is pulling in `dart:html` regardless. This is quite unfortunate, but will likely require some cleanup on the Dart2js side to fix.

Still, definitely something we want to address at some point.

**Review by jakemac53 — APPROVED** [review]

File: pkgs/test/test/runner/browser/code_server.dart (hunk 1 above)
Comment (joshualitt) [reply pkgs/test/test/runner/browser/code_server.dart:40]: Yea, it was just unused. In this case, I was curious how it was used, and noticed there were no references. Grepping also showed no uses.

I'm happy to add it back if you decide it has value.

File: pkgs/test/lib/src/runner/browser/post_message_channel.dart [hunk 3]
```diff
@@ -20,36 +20,42 @@ external void _postParentMessage(Object message, String targetOrigin);
...
-    });
-
-    controller.local.stream.listen((data) {
-      port.postMessage({'data': data});
-    }, onDone: () {
-      port.postMessage({'event': 'done'});
-      portSubscription.cancel();
-    });
-  });
+    dom.MessageEvent message = event as dom.MessageEvent;
```
[the remark this answers is not in the document]
Comment (joshualitt) [reply pkgs/test/lib/src/runner/browser/post_message_channel.dart:33]: Whoops, good catch!

**Pull request merged**